presentation/src/main/java/org/sagebionetworks/assessmentmodel/presentation/AssessmentFragment.kt

```diff
@@ -40,45 +40,57 @@ open class AssessmentFragment : Fragment() {
     }
 
 
-
     lateinit var viewModel: AssessmentViewModel
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        val assessmentId = arguments!!.getString(ARG_ASSESSMENT_ID_KEY)!!
-        val resourceName = arguments!!.getString(ARG_RESOURCE_NAME)!!
-        val packageName = arguments!!.getString(ARG_PACKAGE_NAME)!!
-
         // TODO: syoung 03/10/2020 Move this to a singleton, factory, registry, etc.
-        val fileLoader = FileLoaderAndroid(resources, context?.packageName ?: packageName)
-        val assessmentGroup = AssessmentGroupInfoObject(
-            assessments = listOf(TransformableAssessmentObject(assessmentId, resourceName)),
-            packageName = packageName)
-        val assessmentProvider = FileAssessmentProvider(fileLoader, assessmentGroup, getJsonLoader())
-        viewModel = ViewModelProvider(
-            this, AssessmentViewModelFactory()
-                .create(assessmentId, assessmentProvider))
-            .get(AssessmentViewModel::class.java)
+        initAssessmentViewModel()
         super.onCreate(savedInstanceState) //Needs to be called after viewModel is initialized
     }
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
-                              savedInstanceState: Bundle?): View? {
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
         return inflater.inflate(R.layout.assessment_fragment, container, false)
     }
 
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
 
         viewModel.currentNodeStateLiveData
-                .observe(this.viewLifecycleOwner, Observer<AssessmentViewModel.ShowNodeState>
-                { showNodeState -> this.showStep(showNodeState) })
+            .observe(this.viewLifecycleOwner, Observer<AssessmentViewModel.ShowNodeState>
+            { showNodeState -> this.showStep(showNodeState) })
         viewModel.start()
     }
 
+    open fun initAssessmentViewModel() {
+        val assessmentId = arguments!!.getString(ARG_ASSESSMENT_ID_KEY)!!
+        val resourceName = arguments!!.getString(ARG_RESOURCE_NAME)!!
+        val packageName = arguments!!.getString(ARG_PACKAGE_NAME)!!
+
+        val fileLoader = FileLoaderAndroid(resources, context?.packageName ?: packageName)
+        val assessmentGroup = AssessmentGroupInfoObject(
+            assessments = listOf(TransformableAssessmentObject(assessmentId, resourceName)),
+            packageName = packageName
+        )
+
+        val assessmentProvider =
+            FileAssessmentProvider(fileLoader, assessmentGroup, getJsonLoader())
+        viewModel = ViewModelProvider(
+            this, AssessmentViewModelFactory()
+                .create(assessmentId, assessmentProvider)
+        )
+            .get(AssessmentViewModel::class.java)
```

Comment:
Do you want this pulled out into a helper method you can override?  Then you have less duplicate code than overriding initAssessmentViewModel().